pinot-core/src/main/java/com/linkedin/pinot/core/query/aggregation/groupby/AggregationGroupByTrimmingService.java

```diff
@@ -93,46 +85,75 @@ public AggregationGroupByTrimmingService(@Nonnull AggregationFunction[] aggregat
       return trimmedResults;
     }
 
+    // Trim the result only if the size of intermediateResultsMap is larger than the threshold
     if (intermediateResultsMap.size() > _trimThreshold) {
-      // Need to trim.
-
-      // Construct the priority queues.
-      @SuppressWarnings("unchecked")
-      PriorityQueue<GroupKeyResultPair>[] priorityQueues = new PriorityQueue[_numAggregationFunctions];
-      Object[] sampleResults = intermediateResultsMap.values().iterator().next();
+      // Construct the sorters
+      // For comparable intermediate result, use PriorityQueue as the sorter
+      // For incomparable intermediate result, use TreeMapBasedSorter as the sorter
+      Object[] sorters = new Object[_numAggregationFunctions];
```

Comment:
It's not about overhead. I will try to make them an interface, but that basically means refactoring most of the code... Still, as a private helper class, I don't think that's necessary.